File: example_problems/tutorial/sum/bots/Java/sum_and_difference_mybot.java
import java.util.Scanner;

public class sum_and_difference_mybot {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String line = "";
        while (true) {
            line = scan.nextLine();
	    if(line.charAt(0) == '#') {   // line contains a commented line from the service server
		if(line.equals("# WE HAVE FINISHED")) {
		    System.exit(0);   // exit upon termination of the service server
		}
	    }
	    else {
		int s = Integer.parseInt(line.split(" ")[0]);
		int d = Integer.parseInt(line.split(" ")[1]);
		int x1 = (s + d) / 2;
		int x2 = (s - d) / 2;
		System.out.println(x1 + " " + x2);
            }
        }

    }
}

File: example_problems/tutorial/sum/bots/python/without_library/sum_and_difference_mybot.py
#!/usr/bin/env python3
import math
from sys import stderr, argv

while True:
    line = input()
    #print(f"# BOT: line={line}", file = stderr)
    if line[0] == '#':   # this is a commented line (sent by the service server)
        if '# WE HAVE FINISHED' == line:
            exit(0)   # exit upon termination of the service server
    else:
        s,d = map(int, line.split() )
        x1 = (s + d) // 2
        x2 = (s - d) // 2
        print(f"{x1} {x2}")
